internal/injector/builtin/integration_test.go

```diff
@@ -0,0 +1,95 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package builtin_test
+
+import (
+	"bytes"
+	"errors"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/datadog/orchestrion/internal/injector"
+	"github.com/datadog/orchestrion/internal/injector/builtin"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	samplesDir      string
+	referenceDir    string
+	updateSnapshots bool
+)
+
+func Test(t *testing.T) {
+	t.Parallel()
+
+	dirs, err := os.ReadDir(samplesDir)
+	require.NoError(t, err)
+	for _, dir := range dirs {
+		dir := dir.Name()
+		t.Run(dir, func(t *testing.T) {
+			t.Parallel()
+
+			pkgDir := path.Join(samplesDir, dir)
+
+			tmp := t.TempDir()
+			inj, err := injector.New(pkgDir, injector.Options{
+				Aspects: builtin.Aspects[:],
+				ModifiedFile: func(filename string) string {
+					return path.Join(tmp, path.Base(filename))
+				},
+				PreserveLineInfo: true,
+			})
+			require.NoError(t, err)
+
+			files, err := filepath.Glob(path.Join(pkgDir, "*.go"))
+			require.NoError(t, err)
+			for _, filename := range files {
+				t.Run(path.Base(filename), func(t *testing.T) {
+					res, err := inj.InjectFile(filename, map[string]string{"httpmode": "wrap"})
+					require.NoError(t, err)
```

Comment:
This test is a little setup-heavy (might refactor some of that later), but here is where "the business" happens.